README.md
simple-sidebar
==============
A simple jQuery sidebar plugin

With this simple plug-in you can choose to use the right-sidebar or the left-sidebar and switch from one to the other in just one type! [see options](#options)

## New version: [simplesidebar-v2](http://www.github.com/dcdeiv/simplesidebar-v2)!

###Support:
<a href="https://flattr.com/submit/auto?user_id=dcdeiv&url=http%3A%2F%2Fwww.github.com%2Fdcdeiv%2Fsimple-sidebar" target="_blank"><img src="http://button.flattr.com/flattr-badge-large.png" alt="Flattr this" title="Flattr this" border="0"></a>

##DEMO:
* [DEMO](http://dcdeiv.github.io/simple-sidebar);
* [Right-Sidebar DEMO](http://dcdeiv.github.io/simple-sidebar/demo-right/index.html);
* [Left-Sidebar DEMO](http://dcdeiv.github.io/simple-sidebar/demo-left/index.html);
* [AJAX](http://dcdeiv.github.io/simple-sidebar/demo-ajax/index.html);
* [Material-Designed](http://materialdesigned.tumblr.com/)

##DOWNLOAD
* **Git**: `$ git clone https://github.com/dcdeiv/simple-sidebar`;
* **Bower**: `$ bower install simple-sidebar`.

##CHANGELOG
I strongly recommend to everyone to read the changelog [here](CHANGELOG.md) before updating the plug-in.

##SET UP
Read the guide [here](SETUP-GUIDE.md);

##EXAMPLE OF USAGE:
**These options (except mask and mask.style) are all required**

	$( '#sidebar' ).simpleSidebar({
		settings: {
			opener: '.button',
			wrapper: '#wrapper',
		},
		sidebar: {
			align: 'left'
			width: 250,
			closingLinks: 'a',
			css: {
				zIndex: 100
			}
		},
		mask: {
			css: {
				backgroundColor: 'grey',
				opacity: 0.9,
				filter: 'Alpha(opacity=90)'
			}
		}
	});

#OPTIONS

	//Select the $sidebar. Ex: '.sidebar'
	$( 'sidebar-selector' ).simpleSidebar({
		//Global plug-in settings
		//*REQUIRED*
		settings: {
			//*REQUIRED* OPENER is the selector that fire the animation. It can be an icon a button, an input, an image, a div or a span. Whatever you want. Ex: '.menu-button', '#button'.
			opener: undefined, //STRING
			//*REQUIRED* WRAPPER is the selector for the wrapper/container div/section that holds all content. It is required to animate correctly the document. Fixed elements ("position: fixed") can be included or excluded. It does not matter if you put inside the $sidebar and the navigation-bar nor if you put them outside. Ex: '.wrapper', '#container' etc.
			wrapper: undefined, //STRING
			//IGNORE is the selector for all elements that must be ignored. This feature comes in help especially when you have to install this plug-in in a CMS or blog. TUMBLR Ex: '#ga_target, #tumblr_controls'.
			ignore: undefined, //STRING
			//DATA is the data-attribute for the new elements that the plug-in creates.
			data: 'ssbplugin',
			//ANIMATION properties.
			animation: {
				//DURATION of the animation
				duration: 500,
				//EASING of the animation. If you include the "jQuery-UI" library in your document you can specify an easing (see https://jqueryui.com/resources/demos/effect/easing.html) Ex: 'linear', 'swing', 'easeInQuad', 'easeOutQuad', 'easeInOutQuad', 'easeInCubic', 'easeOutCubic', 'easeInOutCubic', 'easeInQuart', 'easeOutQuart', 'easeInOutQuart'.
				easing: 'swing'
			}
		},
		//*REQUIRED* all SIDEBAR settings
		sidebar: {
			//*REQUIRED* the ALIGN of the $sidebar. Ex: undefined, 'left', 'right'.
			//*WARNING* type in lowercase. If undefined the align is automatically assigned to 'left'.
			align: undefined, //STRING
			//*REQUIRED* the WIDTH and MAXWIDTH of the $sidebar. This value will override CSS.
			width: 350, //NUMBER
			//the GAP between the end of the sidebar (margin-right for the left-sidebar. margin-left for the right-sidebar) and the end of the document (document margin-right for the left-sidebar. document margin-left for the right-sidebar) when the screen is smaller that the $sidebar width.
			gap: 64, //NUMBER
			//CLOSINGLINKS is the selector that will fire the $sidebar closing animation.
			//*WARNING* do not select parent and child. Ex: 'li, a' when "<li><a href=""></a></li>". So just select the child element ('a'). POSSIBLE EXAMPLES: 'a', 'li', 'a, .fauxlinks, #social, .button'.
			closingLinks: 'a',
			//*REQUIRED* This feature assures the plug-in that some values will override CSS.
			css: {
				//*REQUIRED* zINDEX or 'z-index' is required to position the sidebar above ALL content and to position the "mask div" one level below the sidebar.
				zIndex: 3000
				//
				//*WARNING* You can add more style here or in a separate file but everything here will override styles in separate CSS files.
				//*WARNING* Do no care about "position: fixed, top:0, left:0, bottom:0, right:0" because the plug-in will care about it. Positioning rules will be overridden automatically by the plug-in.
			}
		},
		//MASK holds all settings of the mask-div. The mask-div is a required feature that lays over the document content (except the $sidebar) and fires the closing animation function.
		mask: {
			//STYLE holds all CSS rules. Use this feature to stylize the mask.
			css: {
				//Default options.
				backgroundColor: 'black', //if you do not want any color use 'transparent'.
				opacity: 0.5, //if you do not want any opacity use 0.
				filter: 'Alpha(opacity=50)' //IE8 and earlier - If you do not want any opacity use 0.
				//You can add more options.
			}
		},
		//SUBWRAPPER is necessary to let content flow
		subWrapper: { //It should not be necessary to custom this section, however here it is for your pleasure.
			css: {
				position: 'relative',
				height: '100%',
				overflow: 'auto'
			}
		}
	});


CHANGELOG.md
CHANGELOG
============

##[v1.2.2](https://github.com/dcdeiv/simple-sidebar/releases/tag/v1.2.2)
* FIXED:
 * CHANGELOG.md links to versions;
 * Headers.

##[v1.2.0](https://github.com/dcdeiv/simple-sidebar/releases/tag/v1.2.0)
* CHANGED:
 * sidebar.style --> sidebar.css
 * mask.style --> sidebar.css
* ADDED:
 * subWrapper.css
 * subwrapper.css.position: 'relative'

##[v1.1.2](https://github.com/dcdeiv/simple-sidebar/releases/tagv/1.1.2)
* Issue [#5](https://github.com/dcdeiv/simple-sidebar/issues/5)
 * FIXED - scrollbar bug;
 * DELETED - onClose callback.

##[v1.1.1](https://github.com/dcdeiv/simple-sidebar/releases/tag/v/1.1.1)
* Merged [#4](https://github.com/dcdeiv/simple-sidebar/pull/4)
 * ADDED onClose callback;
 * FIXED multiple scrollbars bug.

##[v1.1.0](https://github.com/dcdeiv/simple-sidebar/releases/tag/v1.1.0)
* FIXED:
 * maskDiv double-click bug. 

##[v1.0.3](https://github.com/dcdeiv/simple-sidebar/releases/tag/v1.0.3)
* FIXED:
 * Issue [#3](https://github.com/dcdeiv/simple-sidebar/issues/3).

##[v1.0.2](https://github.com/dcdeiv/simple-sidebar/releases/tag/v1.0.2)
* FIXED:
 * Mask-div: now it is appended and not prepended to body;
 * anchor links: now they work properly;
 * right-sidebar: now it works properly;
* DELETED:
 * colsole.log;
* ADDED:
 * support for elements positioned absolute.

##[v1.0.1](https://github.com/dcdeiv/simple-sidebar/releases/tag/v1.0.1)
* FIXED: maskDiv.


options.js
/*
** Simple Sidebar v1.2.2 by DcDeiv https://github.com/dcdeiv
**
** GPLv2 http://www.gnu.org/licenses/gpl-2.0-standalone.html
**
** # Default Options
** # Example of Usage
**
** - *REQUIRED* means that without that value, the plug-in won't work.
** - *WARNING* means that you have to read very carefully.
** - STRING: "a string", 'another string', 'a "complex" string', "another 'complex' string", '50px', '5em', '100%', '1000ms', etc.
** - NUMBER: 500, 50, 5, 100 - these are all PIXELS or MILLISECONDS(ms).
** - undefined is undefined but not 'undefined' nor "undefined".
**
** Forewarned is forearmed.
**
*/



// # Default Options
====================

//Select the $sidebar. Ex: '.sidebar'
$( 'sidebar-selector' ).simpleSidebar({
	//Global plug-in settings
	//*REQUIRED*
	settings: {
		//*REQUIRED* OPENER is the selector that fire the animation. It can be an icon a button, an input, an image, a div or a span. Whatever you want. Ex: '.menu-button', '#button'.
		opener: undefined, //STRING
		//*REQUIRED* WRAPPER is the selector for the wrapper/container div/section that holds all content. It is required to animate correctly the document. Fixed elements ("position: fixed") can be included or excluded. It does not matter if you put inside the $sidebar and the navigation-bar nor if you put them outside. Ex: '.wrapper', '#container' etc.
		wrapper: undefined, //STRING
		//IGNORE is the selector for all elements that must be ignored. This feature comes in help especially when you have to install this plug-in in a CMS or blog. TUMBLR Ex: '#ga_target, #tumblr_controls'.
		ignore: undefined, //STRING
		//DATA is the data-attribute for the new elements that the plug-in creates.
		data: 'ssbplugin',
		//ANIMATION properties.
		animation: {
			//DURATION of the animation
			duration: 500,
			//EASING of the animation. If you include the "jQuery-UI" library in your document you can specify an easing (see https://jqueryui.com/resources/demos/effect/easing.html) Ex: 'linear', 'swing', 'easeInQuad', 'easeOutQuad', 'easeInOutQuad', 'easeInCubic', 'easeOutCubic', 'easeInOutCubic', 'easeInQuart', 'easeOutQuart', 'easeInOutQuart'.
			easing: 'swing'
		}
	},
	//*REQUIRED* all SIDEBAR settings
	sidebar: {
		//*REQUIRED* the ALIGN of the $sidebar. Ex: undefined, 'left', 'right'.
		//*WARNING* type in lowercase. If undefined the align is automatically assigned to 'left'.
		align: undefined, //STRING
		//*REQUIRED* the WIDTH and MAXWIDTH of the $sidebar. This value will override CSS.
		width: 350, //NUMBER
		//the GAP between the end of the sidebar (margin-right for the left-sidebar. margin-left for the right-sidebar) and the end of the document (document margin-right for the left-sidebar. document margin-left for the right-sidebar) when the screen is smaller that the $sidebar width.
		gap: 64, //NUMBER
		//CLOSINGLINKS is the selector that will fire the $sidebar closing animation.
		//*WARNING* do not select parent and child. Ex: 'li, a' when "<li><a href=""></a></li>". So just select the child element ('a'). POSSIBLE EXAMPLES: 'a', 'li', 'a, .fauxlinks, #social, .button'.
		closingLinks: 'a',
		//*REQUIRED* This feature assures the plug-in that some values will override CSS.
		css: {
			//*REQUIRED* zINDEX or 'z-index' is required to position the sidebar above ALL content and to position the "mask div" one level below the sidebar.
			zIndex: 3000
			//
			//*WARNING* You can add more style here or in a separate file but everything here will override styles in separate CSS files.
			//*WARNING* Do no care about "position: fixed, top:0, left:0, bottom:0, right:0" because the plug-in will care about it. Positioning rules will be overridden automatically by the plug-in.
		}
	},
	//MASK holds all settings of the mask-div. The mask-div is a required feature that lays over the document content (except the $sidebar) and fires the closing animation function.
	mask: {
		//STYLE holds all CSS rules. Use this feature to stylize the mask.
		css: {
			//Default options.
			backgroundColor: 'black', //if you do not want any color use 'transparent'.
			opacity: 0.5, //if you do not want any opacity use 0.
			filter: 'Alpha(opacity=50)' //IE8 and earlier - If you do not want any opacity use 0.
			//You can add more options.
		}
	},
	//SUBWRAPPER is necessary to let content flow
	subWrapper: { //It should not be necessary to custom this section, however here it is for your pleasure.
		css: {
			position: 'relative',
			height: '100%',
			overflow: 'auto'
		}
	}
});



/* # Example of Usage
**======================
** with required options
*/


$( '#sidebar' ).simpleSidebar({
	settings: {
		opener: '.button',
		wrapper: '#wrapper',
	},
	sidebar: {
		align: 'left'
		width: 250,
		closingLinks: 'a',
		css: {
			zIndex: 100
		}
	},
	mask: {
		css: {
			backgroundColor: 'grey',
			opacity: 0.9,
			filter: 'Alpha(opacity=90)'
		}
	}
});
